From: Dmitry Shachnev <mitya57@debian.org>
Date: Sun, 8 Jul 2018 16:15:31 +0300
Subject: Replace snowballstemmer usage with PyStemmer

The snowballstemmer module uses PyStemmer (bindings for libstemmer_c)
when it is available, and falls back to its own pure Python implementation
otherwise.

In Debian the snowballstemmer module is not packaged, so we can use
PyStemmer directly.
---
 Sphinx.egg-info/requires.txt | 1 -
 setup.py                     | 1 -
 sphinx/search/da.py          | 4 ++--
 sphinx/search/de.py          | 4 ++--
 sphinx/search/es.py          | 4 ++--
 sphinx/search/fi.py          | 4 ++--
 sphinx/search/fr.py          | 4 ++--
 sphinx/search/hu.py          | 4 ++--
 sphinx/search/it.py          | 4 ++--
 sphinx/search/nl.py          | 4 ++--
 sphinx/search/no.py          | 4 ++--
 sphinx/search/pt.py          | 4 ++--
 sphinx/search/ro.py          | 4 ++--
 sphinx/search/ru.py          | 4 ++--
 sphinx/search/sv.py          | 4 ++--
 sphinx/search/tr.py          | 4 ++--
 16 files changed, 28 insertions(+), 30 deletions(-)

diff --git a/Sphinx.egg-info/requires.txt b/Sphinx.egg-info/requires.txt
index b645dcc..2e878ce 100644
--- a/Sphinx.egg-info/requires.txt
+++ b/Sphinx.egg-info/requires.txt
@@ -2,7 +2,6 @@ six>=1.5
 Jinja2>=2.3
 Pygments>=2.0
 docutils>=0.11
-snowballstemmer>=1.1
 babel!=2.0,>=1.3
 alabaster<0.8,>=0.7
 imagesize
diff --git a/setup.py b/setup.py
index b6b3bc2..7c1e03d 100644
--- a/setup.py
+++ b/setup.py
@@ -20,7 +20,6 @@ install_requires = [
     'Jinja2>=2.3',
     'Pygments>=2.0',
     'docutils>=0.11',
-    'snowballstemmer>=1.1',
     'babel>=1.3,!=2.0',
     'alabaster>=0.7,<0.8',
     'imagesize',
diff --git a/sphinx/search/da.py b/sphinx/search/da.py
index eb7394b..29d8f78 100644
--- a/sphinx/search/da.py
+++ b/sphinx/search/da.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -131,7 +131,7 @@ class SearchDanish(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('danish')
+        self.stemmer = Stemmer.Stemmer('danish')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/de.py b/sphinx/search/de.py
index 90fc4d0..547cf28 100644
--- a/sphinx/search/de.py
+++ b/sphinx/search/de.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -314,7 +314,7 @@ class SearchGerman(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('german')
+        self.stemmer = Stemmer.Stemmer('german')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/es.py b/sphinx/search/es.py
index 6fda9ca..2987255 100644
--- a/sphinx/search/es.py
+++ b/sphinx/search/es.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -374,7 +374,7 @@ class SearchSpanish(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('spanish')
+        self.stemmer = Stemmer.Stemmer('spanish')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/fi.py b/sphinx/search/fi.py
index bf757d5..bb3a29f 100644
--- a/sphinx/search/fi.py
+++ b/sphinx/search/fi.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -124,7 +124,7 @@ class SearchFinnish(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('finnish')
+        self.stemmer = Stemmer.Stemmer('finnish')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/fr.py b/sphinx/search/fr.py
index 9976c1c..497858b 100644
--- a/sphinx/search/fr.py
+++ b/sphinx/search/fr.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -210,7 +210,7 @@ class SearchFrench(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('french')
+        self.stemmer = Stemmer.Stemmer('french')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/hu.py b/sphinx/search/hu.py
index 03cbf8c..e667387 100644
--- a/sphinx/search/hu.py
+++ b/sphinx/search/hu.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -238,7 +238,7 @@ class SearchHungarian(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('hungarian')
+        self.stemmer = Stemmer.Stemmer('hungarian')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/it.py b/sphinx/search/it.py
index 860ab93..9425679 100644
--- a/sphinx/search/it.py
+++ b/sphinx/search/it.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -327,7 +327,7 @@ class SearchItalian(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('italian')
+        self.stemmer = Stemmer.Stemmer('italian')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/nl.py b/sphinx/search/nl.py
index de4fd13..748d31f 100644
--- a/sphinx/search/nl.py
+++ b/sphinx/search/nl.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -138,7 +138,7 @@ class SearchDutch(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('dutch')
+        self.stemmer = Stemmer.Stemmer('dutch')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/no.py b/sphinx/search/no.py
index 81876bc..db28104 100644
--- a/sphinx/search/no.py
+++ b/sphinx/search/no.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -213,7 +213,7 @@ class SearchNorwegian(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('norwegian')
+        self.stemmer = Stemmer.Stemmer('norwegian')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/pt.py b/sphinx/search/pt.py
index 9afe808..ab48ee7 100644
--- a/sphinx/search/pt.py
+++ b/sphinx/search/pt.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -273,7 +273,7 @@ class SearchPortuguese(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('portuguese')
+        self.stemmer = Stemmer.Stemmer('portuguese')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/ro.py b/sphinx/search/ro.py
index 69d6edd..31a585d 100644
--- a/sphinx/search/ro.py
+++ b/sphinx/search/ro.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -32,7 +32,7 @@ class SearchRomanian(SearchLanguage):
 
     def init(self, options):
         # type: (Dict) -> None
-        self.stemmer = snowballstemmer.stemmer('romanian')
+        self.stemmer = Stemmer.Stemmer('romanian')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/ru.py b/sphinx/search/ru.py
index cc189a9..e21726b 100644
--- a/sphinx/search/ru.py
+++ b/sphinx/search/ru.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -262,7 +262,7 @@ class SearchRussian(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('russian')
+        self.stemmer = Stemmer.Stemmer('russian')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/sv.py b/sphinx/search/sv.py
index fdb6468..fde58cd 100644
--- a/sphinx/search/sv.py
+++ b/sphinx/search/sv.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage, parse_stop_word
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -150,7 +150,7 @@ class SearchSwedish(SearchLanguage):
 
     def init(self, options):
         # type: (Any) -> None
-        self.stemmer = snowballstemmer.stemmer('swedish')
+        self.stemmer = Stemmer.Stemmer('swedish')
 
     def stem(self, word):
         # type: (unicode) -> unicode
diff --git a/sphinx/search/tr.py b/sphinx/search/tr.py
index 1b8a0f9..2e8c1f0 100644
--- a/sphinx/search/tr.py
+++ b/sphinx/search/tr.py
@@ -11,7 +11,7 @@
 
 from sphinx.search import SearchLanguage
 
-import snowballstemmer
+import Stemmer
 
 if False:
     # For type annotation
@@ -32,7 +32,7 @@ class SearchTurkish(SearchLanguage):
 
     def init(self, options):
         # type: (Dict) -> None
-        self.stemmer = snowballstemmer.stemmer('turkish')
+        self.stemmer = Stemmer.Stemmer('turkish')
 
     def stem(self, word):
         # type: (unicode) -> unicode
